strawberry/types/types.py

```diff
@@ -167,3 +184,30 @@ def is_implemented_by(self, root: Union[type, dict]) -> bool:
 
         # All field mappings succeeded. This is a match
         return True
+
+
+class WithStrawberryDefinition(Protocol):
+    __strawberry_definition__: StrawberryObjectDefinition
+
+
+def has_strawberry_definition(klass: Any) -> TypeGuard[Type[WithStrawberryDefinition]]:
```

Comment:
Would prefer `obj` as opposed to `klass` as these functions can be called on any object, not just classes:
https://docs.python.org/3/reference/datamodel.html#data-model

Reply:
Also I believe `WithStrawberryDefinition` and these functions should be moved to `strawberry/type.py` with the latter becoming methods defined on `StrawberryType`. It would also avoid having to perform the deferred import of this module there. Do you agree?

Reply:
Since this method would in the future be true only for object type (input / interface / object) I putted it there. But I think you are right fits better there!

Reply:
yeah, let's move it to strawberry/type.py

I want to get rid of strawberry/types/types.py at some point :D